### PR TITLE
feat: fast checkout payment (e.g. PAYPAL Express)

### PR DESCRIPTION
**Pull request by skoch-intershop**
## PR Type

[x] Feature

## What Is the Current Behavior?

Not possible to apply fast checkout payment methods.

Issue Number: Closes #95945 (ADO)

## What Is the New Behavior?

The fast checkout payment functionality is now available on the basket detail page, if the user is logged in or the 'guestCheckout' feature toggle is switched on.
All available fast checkout payment methods will be displayed in the summary slot .
image

## Does this PR Introduce a Breaking Change?
[x] No

## Notes for the Changelog

CHANGES THAT REQUIRE MORE RECENT ICM VERSIONS
fast checkout payment (#1682) - ICM 12.0.0 or ICM 7.10.41.3

## Other

[AB#97352](https://dev.azure.com/intershop-com/cefd1005-00a7-4c79-927f-a16947d1b2e6/_workitems/edit/97352)

### CODE REVIEW

File: src/app/pages/basket/shopping-basket-payment/shopping-basket-payment.component.ts [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+  fastCheckout(paymentId: string) {
+    this.checkoutFacade.setFastCheckoutPayment(paymentId);
+  }
+
+  /**
+   * Determine whether payment cost threshold has been reached
+   * for usage in template
+   */
+  paymentCostThresholdReached(paymentMethod: PaymentMethod): boolean {
```
Comment (SGrueber) [inline src/app/pages/basket/shopping-basket-payment/shopping-basket-payment.component.ts:49]: don't copy code - if you need the same functionality here as in the checkout-payment.component create a subcomponent or find another way to share the code

**Review by SGrueber — CHANGES_REQUESTED** [review]

File: src/app/core/models/payment-method/payment-method.mapper.ts [hunk 2]
```diff
@@ -103,7 +103,7 @@ export class PaymentMethodMapper {
    * valid: payment methods without capabilities or which have no capabilities given in the list below
    */
   private static isPaymentMethodValid(paymentData: PaymentMethodBaseData): boolean {
-    const invalidCapabilities = ['LimitedTender', 'FastCheckout'];
+    const invalidCapabilities = ['LimitedTender'];
```
Comment (SGrueber) [inline src/app/core/models/payment-method/payment-method.mapper.ts:106]: getEligiblePaymentMethods now returns also fast checkout methods, pls write a migration note, that the selector has changed

File: src/app/core/store/customer/basket/basket.actions.ts [hunk 3]
```diff
@@ -248,6 +248,16 @@ export const setBasketPaymentFail = createAction('[Basket API] Set a Payment at
 
 export const setBasketPaymentSuccess = createAction('[Basket API] Set a Payment at Basket Success');
 
+export const setFastCheckoutPayment = createAction(
```
Comment (SGrueber) [inline src/app/core/store/customer/basket/basket.actions.ts:251]: rename to setBasketFastCheckoutPayment

File: src/app/core/store/customer/basket/basket-payment.effects.ts [hunk 4]
```diff
@@ -34,7 +39,50 @@ import { getCurrentBasket, getCurrentBasketId } from './basket.selectors';
...
+    private actions$: Actions,
+    private store: Store,
+    private paymentService: PaymentService,
+    private basketService: BasketService
+  ) {}
+
+  /**
+   * The redirect after fast checkout payment method effect.
+   */
+  executeFastCheckoutRedirect$ = createEffect(
```
Comment (SGrueber) [inline src/app/core/store/customer/basket/basket-payment.effects.ts:52]: test ? see order.effects.spec

**Review by SGrueber — CHANGES_REQUESTED** [review]
it looks all much better than before 👍

**Review by iwiederhold — COMMENTED** [review]
Please remove the space of 16px after the button-group. (B2B+B2C)

image

For a better usability we should insert a space of 1rem between the checkout button and the other ones.

image

**Review by M-Behr — CHANGES_REQUESTED** [review]
Paypal_express_error_payment_canceled

Please use a toast error message instead of this inline error message. Please create a new text for this error because the user does not see the payment methods here.

Error message for the cart page: "We're sorry. Your payment was canceled. You can continue with the checkout or use the express checkout."

**Comment by M-Behr** [comment]
> Please remove the space of 16px after the button-group. (B2B+B2C)
> 
> image
> 
> For a better usability we should insert a space of 1rem between the checkout button and the other ones. There used to be an "or" there. That was even better.
> 
> image

We decided to show the "or" between checkout and express checkout button, exactly as it is in the Responsive Starter Store.

**Pull request merged**